server/publications/channelAutocomplete.coffee

```diff
@@ -13,15 +13,12 @@ Meteor.publish 'channelAutocomplete', (name) ->
 		limit: 5
 
 	cursorHandle = RocketChat.models.Rooms.findByNameContainingAndTypes(name, ['c'], options).observeChanges
-		added: (_id, record) ->
-			pub.added('channel-autocomplete', _id, record)
-
+        added: (_id, record) ->
+			pub.added('channel-autocomplete', _id, record) if _id?
```

Comment:
@geekgonecrazy  This PR added Chinese characters support for channel and group names.
I removed slugify function as it would be impossible for it to deal with Chinese Characters.

I knew it's not a generic solution for Unicode Character support, it works though.
I am not sure if it's suitable to accept all utf-8 codes.  I didn't see the barrier as  Browsers can escape it in the URL.